src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -437,51 +468,120 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                 return;
             }
 
+            StepListener<GetModelResponse> getModelStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> blockModelIdStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> clearModelMetadataStep = new StepListener<>();
+            StepListener<DeleteResponse> deleteModelFromIndexStep = new StepListener<>();
+            StepListener<RemoveModelFromCacheResponse> clearModelFromCacheStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> unblockModelIdStep = new StepListener<>();
+
+            // Get Model to check if model is in TRAINING
+            get(modelId, ActionListener.wrap(getModelStep::onResponse, exception -> {
+                if (exception instanceof ResourceNotFoundException) {
+                    listener.onResponse(new DeleteModelResponse(modelId, "failed", exception.getMessage()));
+                    return;
+                }
+                listener.onFailure(exception);
+            }));
+
+            getModelStep.whenComplete(getModelResponse -> {
+                // If model is in Training state, fail delete model request
+                if (ModelState.TRAINING.equals(getModelResponse.getModel().getModelMetadata().getState())) {
+                    String errorMessage = String.format("Cannot delete model \"%s\". Model is still in training", modelId);
+                    logger.error(errorMessage);
```

Comment:
I dont think this needs to be logged as an error, but would be interested to hear from @navneet1v on this.

Reply:
I agree with Jack, it doesn't seem like an error, should be something less critical like info maybe. Error would have more sense it if try to delete a request and then fail with some runtime error like IO problem etc.

Reply:
ack